src/components/Card.js

```diff
@@ -23,13 +23,70 @@ export default class Card extends Component {
         twitterIcon: <i className="fab fa-twitter-square" />,
         close: <i className="fas fa-times-circle" /> 
       },
+      favorite: false
     };
   }
+
+  componentDidMount() {
+    this.checkFavorite();
+  }
+
+  checkFavorite = () => {
+    if (localStorage.getItem("favoritesList")){
+      let localFavoritesList = JSON.parse(localStorage.getItem("favoritesList")).map(object => object.name);
+      if (localFavoritesList.includes(this.props.cardData.name)) {
+        this.setState({
+          favorite: true
+        })
+      }
+    } 
+  }
+
+  handleFavorites = () => {
+    if (this.state.favorite) {
+      this.removeFromFavorites();
+    } else {
+      this.addToFavorites();
+    }
+    this.setState({
+      favorite: !this.state.favorite
+    });
+  }
+
+  removeFromFavorites = () => {
+    let workingFavoritesList = JSON.parse(localStorage.getItem("favoritesList"));
+    let targetIndex = workingFavoritesList.map(fave => fave.name).indexOf(this.props.cardData.name);
+    workingFavoritesList.splice(targetIndex, 1);
+    localStorage.setItem("favoritesList", JSON.stringify(workingFavoritesList));
+    this.props.updateFavorites(workingFavoritesList);
+  }
+
+  addToFavorites = () => {
+    let workingFavoritesList;
+    if (this.props.favoritesList) {
+      workingFavoritesList = JSON.parse(localStorage.getItem("favoritesList"));
+      workingFavoritesList.push(this.props.cardData);
+      localStorage.setItem("favoritesList", JSON.stringify(workingFavoritesList));
+    } else {
+      workingFavoritesList = [this.props.cardData];
+      localStorage.setItem("favoritesList", JSON.stringify(workingFavoritesList));
+    }
+    this.props.updateFavorites(workingFavoritesList);
+  }
+
   
   render() {
     const cardData = this.props.cardData;
     const { id } = cardData;
     let mainInfo, cardType;
+    let favorite = false;
+    
+    if (localStorage.getItem("favoritesList")) {
+      let localFavoritesList = JSON.parse(localStorage.getItem("favoritesList")).map(object => object.name);
+      if (localFavoritesList.includes(this.props.cardData.name)) {
+          favorite = true
+      }
```

Comment:
Would rounding off this conditional with an ` else favorite = false ` help us fix the bug where the heart stays red switching from card to card?
@jacobogart

Reply:
Right now there is both a favorite state that determines if the element needs to be added or removed and a local favorites variable that determines if the heart should be red or not. This logic just evaluates if the heart should be red, which is set at false at the top of the render. But I guess an else wouldn't hurt.